src/parquet/file_writer.h

```diff
@@ -79,6 +83,44 @@ class PARQUET_EXPORT RowGroupWriter {
   std::unique_ptr<Contents> contents_;
 };
 
+// RowGroupWriter implementation that allows bounding a RowGroup size to a certain value
+// Columns can be written in an order
+// All the values are compressed and stored in memory
+// Values are written to the file on Close
+
+class PARQUET_EXPORT RowGroupWriter2 {
```

Comment:
A name like "FixedSizeRowGroupWriter", really anything other than <something>2, would be a little more intuitive in the public API.

Reply:
Agreed about the name. Something like ContinuousRowGroupWriter might work a bit better though since we now allow writing to each column multiple times, and this way suffix "2" can be replaced in all other locations with something like "_cont". "Repeated" might work too.

Reply:
Agree that `RowGroupWriter2` is not a good name. 

Why do we need a new class? It's not clear to me that it's needed

Reply:
The current `RowGroupWriter` class has `virtual ColumnWriter* NextColumn() = 0;` which does not fit well with the use case here which is to be able to `write to a column chunk any number of times (can cycle around all column chunks any number of times) and also any order`. 
The current `RowGroupWriter` requires us to write a column chunk at one go and move to the immediate next column. 
We need the following external API to support this use case.
`virtual ColumnWriter* get_column(i) = 0;`
The new `RowGroupWriter2` has different semantics in this aspect.

Reply:
I am not tied to this name. Will change to `FixedSizeRowGroupWriter`